docs/DeveloperGuide.md

```diff
@@ -1,34 +1,851 @@
+1. [Appendix: Requirements](#6)
+    <br/>&nbsp;6.1 [Product Scope](#6.1)
+    <br/>&nbsp;6.2 [User Stories](#6.2)
+    <br/>&nbsp;6.3 [Use Cases](#6.3)
+    <br/>&nbsp;6.4 [Non-Functional Requirements](#6.4)
+    <br/>&nbsp;6.5 [Glossary](#6.5)
+1. [Appendix: Instructions for Manual Testing](#7)
+
+## 1. Introduction <a name="1"></a>
+_ModTracker_ is a desktop app for NUS students to track the time spent 
+as well as tasks to do for each of their modules.
+It helps students prioritise their work and 
+balance their time spent amongst their modules. 
+This app uses a Command Line Interface (CLI) and is written in Java 11.
+
+## 2. Setting Up, Getting Started <a name="2"></a>
+
+### 2.1 Setting up the project in your local machine <a name="2.1"></a>
+Ensure that you have JDK 11 or above installed on your computer. 
+
+First, **fork** this [repo](https://github.com/AY2021S1-CS2113T-F12-4/tp), and **clone** a copy to your computer by using any Git client.
+
+If you plan to use IntelliJ IDEA (highly recommended):
+1. Ensure IntelliJ is configured to use JDK 11.
+    * Start up IntelliJ into the welcome page.
+    * Click on `Configure` -> `Structure for New Projects` -> `Project Settings` -> `Project`.
+    * Under the `Project SDK:` section, ensure that the correct SDK (**java version "11.0.8"** or higher) is selected.
+1. Import the project as a Gradle project.
+    * At the same welcome page, click on `Open or Import`.
+    * Navigate to the folder where you cloned the repo, locate the `build.gradle` file within the folder and select it.
+    * Choose the `Open as Project` option when asked.
+    * The project may take a couple of minutes to import depending on your actual computer hardware setup.
+1. Verify the setup.
+    * Under the `seedu.modtracker` package, locate the `ModTracker` class and run it.
+    * Try a few commands. You may want to refer to the [user guide](https://ay2021s1-cs2113t-f12-4.github.io/tp/UserGuide.html).
+    * Run the [tests]() to ensure they all pass.
+
+### 2.2 Before writing the code <a name="2.2"></a>
+1. Configure the coding style
+    * If using IntelliJ IDEA, follow this [guide](https://se-education.org/guides/tutorials/intellijCodeStyle.html) to set up IDEA's coding style yo match ours.
+    
+1. Set up CI
+    * This project comes with a GitHub Actions config files (in `.github/workflows` folder). When GitHub detects those files, 
+    it will run the CI for the project automatically at each push to the `master` branch or to any PR.
+    No setup is required for this.
+    
+1. Learn the design
+    * When you are ready to start coding, we recommend that you look at [ModTracker's architecture](#architecture) to get some 
+    sense of the overall design.
+
+
+## 3. Design <a name="3"></a>
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### 3.1 Architecture <a name="3.1"></a>
+![Architecture](diagrams/Architecture.png)
```

Comment:
Is the diagram incomplete? It seems to be missing some labels and has a few alignment issues.